src/aeb_fate_generate_ops.erl

```diff
@@ -0,0 +1,345 @@
+    , {'BITS_NONEA',    16#6e,    0,  false,   3, atomic,        bits_none, "accumulator := empty bitmap."}
+    , {'BITS_NONE',     16#36,    1,  false,   3, [a],           bits_none, "Arg0 := empty bitmap."}
+    , {'BITS_ALLA',     16#6f,    0,  false,   3, atomic,         bits_all, "accumulator := full bitmap."}
+    , {'BITS_ALL',      16#37,    1,  false,   3, [a],            bits_all, "Arg0 := full bitmap."}
+    , {'BITS_ALL_N',    16#6d,    2,  false,   3, [a,a],        bits_all_n, "Arg0 := bitmap with Arg1 bits set."}
+    , {'BITS_SET',      16#38,    3,  false,   3, [a,a,a],        bits_set, "Arg0 := set bit Arg2 of bitmap Arg1."}
+    , {'BITS_CLEAR',    16#39,    3,  false,   3, [a,a,a],      bits_clear, "Arg0 := clear bit Arg2 of bitmap Arg1."}
+    , {'BITS_TEST',     16#3a,    3,  false,   3, [a,a,a],       bits_test, "Arg0 := true if bit Arg2 of bitmap Arg1 is set."}
+    , {'BITS_SUM',      16#3b,    2,  false,   3, [a,a],          bits_sum, "Arg0 := sum of set bits in bitmap Arg1. Exception if infinit bitmap."}
+    , {'BITS_OR',       16#3c,    3,  false,   3, [a,a,a],         bits_or, "Arg0 := Arg1 v Arg2."}
+    , {'BITS_AND',      16#3d,    3,  false,   3, [a,a,a],        bits_and, "Arg0 := Arg1 ^ Arg2."}
+    , {'BITS_DIFF',     16#3e,    3,  false,   3, [a,a,a],       bits_diff, "Arg0 := Arg1 - Arg2."}
+    , {'DUPA',          16#70,    0,  false,   3, atomic,              dup, "push copy of accumulator on stack."}
+    , {'DUP',           16#0a,    1,  false,   3, [a],                 dup, "push Arg0 stack pos on top of stack."}
+    , {'POP',           16#0b,    1,  false,   3, [a],                 pop, "Arg0 := top of stack."}
+    , {'STORE',         16#10,    2,  false,   3, [a,a],             store, "Arg0 := Arg1."}
+    , {'ADDRESS',       16#3f,    1,  false,   3, [a],             address, "Arg0 := The current contract address."}
+    , {'BALANCE',       16#40,    1,  false,   3, [a],             balance, "Arg0 := The current contract address."}
+    , {'ORIGIN',        16#41,    1,  false,   3, [a],              origin, "Arg0 := Address of contract called by the call transaction."}
+    , {'CALLER',        16#42,    1,  false,   3, [a],              caller, "Arg0 := The address that signed the call transaction."}
+    , {'GASPRICE',      16#43,    1,  false,   3, [a],            gasprice, "Arg0 := The current gas price."}
+    , {'BLOCKHASH',     16#44,    1,  false,   3, [a],           blockhash, "Arg0 := The current blockhash."} %% TODO: Do we support has at height?
+    , {'BENEFICIARY',   16#45,    1,  false,   3, [a],         beneficiary, "Arg0 := The address of the current beneficiary."}
+    , {'TIMESTAMP',     16#46,    1,  false,   3, [a],           timestamp, "Arg0 := The current timestamp. Unrelaiable, don't use for anything."}
+    , {'NUMBER',        16#47,    1,  false,   3, [a],              number, "Arg0 := The block height."}
```

Comment:
NUMBER is a very generic name...

Reply:
Yes its the name used by EVM. I can change it in fate. We could have blockheight and generation instead.

Reply:
Or MICROBLOCK and GENERATION might be better.